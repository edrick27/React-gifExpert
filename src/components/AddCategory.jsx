import { useState } from "react"
import { PropTypes } from "prop-types";

export const AddCategory = ({ onAddNewCategory }) => {

    const [inputValue, setInputValue] = useState('juan caballo')

    const onInputChange = ({ target }) => {
        setInputValue(target.value)
    }

    const onSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length < 1) return;

        setInputValue('');
        onAddNewCategory(inputValue.trim());
    }

    return (
        <>
            <form onSubmit={ onSubmit } aria-label="form">
                <input
                    type="text"
                    placeholder="Buscar Gifts..."
                    value={inputValue}
                    onChange={onInputChange}
                />
                <button></button>
            </form>
        </>
    )
}

AddCategory.propTypes = {
    onAddNewCategory: PropTypes.func.isRequired
}
