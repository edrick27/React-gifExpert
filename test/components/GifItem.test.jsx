import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem"

describe('Prueba del GifItem', () => {

    const title = "hola Mundo";
    const url = "https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=a2ccfd36agv0f80hafuikwpxbss1yllzprvgemxszpif1mrs&ep=v1_gifs_search&rid=giphy.gif&ct=g"

    test('Snapshot del GifItem', () => {

        const { container } = render(<GifItem title={ title } url={ url } />);

        expect(container).toMatchSnapshot();
    })

    test('debe mostrar el title y el url en el lugar indicado', () => {

        render(<GifItem title={ title } url={ url } />);

        const img = screen.getByRole('img').src;
        expect(img).toBe(url);
    })

    test('El titulo debe estar en el lugar indicado', () => {

        render(<GifItem title={ title } url={ url } />);
        expect(screen.getByText(title)).toBeTruthy();
    })
})