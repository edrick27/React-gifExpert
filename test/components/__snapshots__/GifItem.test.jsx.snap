// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del GifItem Snapshot del GifItem 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="hola Mundo"
      src="https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=a2ccfd36agv0f80hafuikwpxbss1yllzprvgemxszpif1mrs&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      hola Mundo
    </p>
  </div>
</div>
`;
